scripts/upgrade

```diff
@@ -12,41 +12,75 @@ source /usr/share/yunohost/helpers
 #=================================================
 # LOAD SETTINGS
 #=================================================
-ynh_script_progression --message="Loading installation settings..." --time --weight=1
+ynh_script_progression --message="Loading installation settings..." --weight=1
 
 app=$YNH_APP_INSTANCE_NAME
 
 domain=$(ynh_app_setting_get --app=$app --key=domain)
 path_url=$(ynh_app_setting_get --app=$app --key=path)
+port=$(ynh_app_setting_get --app=$app --key=port)
+port_wg=$(ynh_app_setting_get --app=$app --key=port_wg)
 final_path=$(ynh_app_setting_get --app=$app --key=final_path)
 
 #=================================================
 # CHECK VERSION
 #=================================================
 
-### This helper will compare the version of the currently installed app and the version of the upstream package.
-### $upgrade_type can have 2 different values
-### - UPGRADE_APP if the upstream app version has changed
-### - UPGRADE_PACKAGE if only the YunoHost package has changed
-### ynh_check_app_version_changed will stop the upgrade if the app is up to date.
-### UPGRADE_APP should be used to upgrade the core app only if there's an upgrade to do.
 upgrade_type=$(ynh_check_app_version_changed)
 
 #=================================================
 # ENSURE DOWNWARD COMPATIBILITY
 #=================================================
-ynh_script_progression --message="Ensuring downward compatibility..." --time --weight=1
+ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
 
 # If final_path doesn't exist, create it
 if [ -z "$final_path" ]; then
 	final_path=/opt/yunohost/$app
 	ynh_app_setting_set --app=$app --key=final_path --value=$final_path
 fi
 
+# Change port for WireGuard UI because 8095 is default port for WireGuard VPN
+if [ $port -eq 8095 ]
+then
+	ynh_app_setting_delete --app=$app --key=port
+	port=$(ynh_find_port --port=8096)
+	ynh_app_setting_set --app=$app --key=port --value=$port
+	ynh_app_setting_set --app=$app --key=is_public --value=0
+fi
+
+# Downgrade linux-image-$arch if updated to the buster-backports version
+arch=$(ynh_detect_arch)
+linuximage_version=$(ynh_package_version --package=linux-image-$arch)
+if [[ $linuximage_version == *"bpo10"* ]]
+then
+	# Downgrading using ynh_package_install apt command without "--no-remove" and with "--allow-downgrades"
+	# It will remove wireguard-ynh-deps and wireguard but they will be reinstalled throught upgrade process
+	ynh_apt --allow-downgrades --option Dpkg::Options::=--force-confdef \
+        --option Dpkg::Options::=--force-confold install linux-image-$arch/stable
+
+	#Remove backports kernel if running on it and send a mail to the admin to ask him to reboot
+	linuxkernel_version=$(uname -r)
+	if [[ $linuxkernel_version == *"bpo"* ]]
+	then
+		ynh_package_remove linux-image-$linuxkernel_version
```

Comment:
:thinking: :thinking: :thinking: 

This sounds like something very touchy ?

:thinking: :thinking: :thinking:

Reply:
Yep, the  version before #5 was installing the backports kernel, now we use the DKMS module, so we need to get it removed to return on the stable kernel. And yes, very touchy.

Note: if the user had it installed and doesn't run on it, it doesn't get remove

Reply:
We can remove ynh_package_remove and just send a email about the fact that the user use the backports kernel, if it's needed

Reply:
I have no opinion on this. What do you advise @alexAubin?

Reply:
Hmpf idk, not sure to understand all the implications, naively considering the app was not flagged as "working" yet we should not worry that much about what happens in first iterations .. So i suppose just sending an email should be good enough ?

Reply:
Then let's remove the package removal and change the content of the email to tell the user about the backports kernel.
We shall maybe warn the user about the lack of automatic update of this kernel.

Reply:
Note: the upgrade from commit check is useless, if we apply this